src/Parser/Hunter/BeastMastery/Modules/Spells/DireBeast/DireBeast.js

```diff
@@ -0,0 +1,122 @@
+
+class DireBeast extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  casts = 0;
+  effectiveBWReduction = 0;
+  wastedBWReduction = 0;
+  bestialWrathCast = 0;
+  badDBCasts = 0;
+  DireBeastCastsSinceLastBW = 0;
+  remainingBestialWrathCooldown = 0;
+
+  on_initialized() {
+    this.active = !this.combatants.selected.hasTalent(SPELLS.DIRE_FRENZY_TALENT.id);
+  }
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.DIRE_BEAST.id && spellId !== SPELLS.BESTIAL_WRATH.id) {
+      return;
+    }
+    if (this.combatants.selected.hasBuff(SPELLS.HUNTER_BM_T19_4P_BONUS.id)) {
+      COOLDOWN_REDUCTION_MS = 16000;
+    }
+    if (spellId === SPELLS.DIRE_BEAST.id) {
+      this.casts += 1;
+      const bestialWrathIsOnCooldown = this.spellUsable.isOnCooldown(SPELLS.BESTIAL_WRATH.id);
+      if (bestialWrathIsOnCooldown) {
+        this.remainingBestialWrathCooldown = this.spellUsable.cooldownRemaining(SPELLS.BESTIAL_WRATH.id);
+        const reductionMs = this.spellUsable.reduceCooldown(SPELLS.BESTIAL_WRATH.id, COOLDOWN_REDUCTION_MS);
+        if (this.remainingBestialWrathCooldown > CD_ON_BESTIAL_WRATH_BAD_DB_THRESHHOLD) {
+          this.effectiveBWReduction += reductionMs;
+          this.wastedBWReduction += (COOLDOWN_REDUCTION_MS - reductionMs);
+        } else {
+          this.badDBCasts += 1;
+          this.effectiveBWReduction += reductionMs;
+          this.wastedBWReduction += (COOLDOWN_REDUCTION_MS - reductionMs);
+        }
+      } else {
+        this.wastedBWReduction += COOLDOWN_REDUCTION_MS;
+        this.badDBCasts += 1;
+      }
+      this.DireBeastCastsSinceLastBW += 1;
+    }
+    if (spellId === SPELLS.BESTIAL_WRATH.id) {
+      this.bestialWrathCast = event.timestamp;
+      this.DireBeastCastsSinceLastBW = 0;
+    }
+  }
+
+  statistic() {
+    const gainedBestialWraths = this.effectiveBWReduction / BESTIAL_WRATH_BASE_CD;
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.DIRE_BEAST.id} />}
+        value={(
+          <span>
+            {`gained `}{formatNumber(gainedBestialWraths, 1)}{'  '}
+            <SpellIcon
+              id={SPELLS.BESTIAL_WRATH.id}
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+              }}
+            />
+            <br />
+            {this.casts}{'  '}
+            <SpellIcon
+              id={SPELLS.DIRE_BEAST.id}
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+              }}
+            />
+            {'  '}
+            {this.badDBCasts}{'  '}
+            <SpellIcon
+              id={SPELLS.DIRE_BEAST.id}
+              style={{
+                height: '1.3em',
+                marginTop: '-.1em',
+                filter: 'grayscale(100%)',
+              }}
+            />
+          </span>
+        )}
+        label={`Direbeast information`}
```

Comment:
It looks again like you've clumped several semi related stats into the same box. This has forced you to hide stat meaning in the tooltip, which reduces readability.

You should probably decide which of these stats is the most important, and then keep only it while putting the other stats in the tooltip. Alternately, if they're all really important, you could put them in seperate stat boxes. 

The problem is that if they're so unrelated that the best label you can come up with is "Direbeast information", they probably shouldn't be together.

Reply:
I'll split it into two modules, to promote clarity